File: src/dotnet/Ansight.Tools.VisualTree/GetVisualTreeTool.cs
namespace Ansight.Tools.VisualTree;

public sealed class GetVisualTreeTool : ITool
{
    public string Category => "ui";

    public ToolScope Scope => ToolScope.Read;

    public string Id => VisualTreeToolIds.GetVisualTree;

    public string Name => "Get Visual Tree";

    public string Description => "Returns the current UI hierarchy for the requested visual-tree source.";

    public string Keywords => "ui visual tree hierarchy layout";

    public ToolSchema ArgumentsSchema => VisualTreeToolSchemas.GetVisualTreeArguments;

    public ToolSchema ResultSchema => VisualTreeToolSchemas.VisualTreeResult;

    public ToolSecurity Security => VisualTreeToolSecurityProfiles.GetVisualTree;

    public async Task<ToolResult> Execute(IReadOnlyDictionary<string, string> arguments)
    {
        ArgumentNullException.ThrowIfNull(arguments);
        arguments.TryGetValue("source", out var source);
        return await VisualTreeSnapshotStore.CaptureAsync(source, arguments);
    }
}

File: src/ios/Sources/AnsightToolsVisualTree/GetVisualTreeTool.swift
import AnsightCore
import Foundation

public final class GetVisualTreeTool: AnsightTool {
    public init() {}

    public var descriptor: AnsightToolDescriptor {
        AnsightToolDescriptor(
            id: AnsightVisualTreeToolIds.getVisualTree,
            name: "Get Visual Tree",
            description: "Returns the current UI hierarchy for the foreground scene.",
            category: "ui",
            policy: .read,
            keywords: "ui visual tree hierarchy layout",
            argumentsSchema: AnsightVisualTreeToolSchemas.getVisualTreeArguments,
            resultSchema: AnsightVisualTreeToolSchemas.visualTreeResult
        )
    }

    public func execute(arguments: [String: String]) throws -> AnsightToolExecutionResult {
        AnsightVisualTreeSnapshotStore.capture(source: arguments["source"], arguments: arguments)
    }
}
